### PR TITLE
Add license information and reuse linter

### PR DESCRIPTION
**Pull request by avnik**

### CODE REVIEW

**Comment by aciceri** [comment]
Please add the the license also to the docs page in the website.

**Comment by avnik** [comment]
@aciceri fixed

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: cardano.nix
+Upstream-Contact: <https://github.com/mlabs-haskell/cardano.nix>
```
Comment (aciceri) [inline .reuse/dep5:3]: Please change contact to `legal@mlabs.city`

File: .reuse/dep5 [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+Source: https://github.com/mlabs-haskell/cardano.nix
+
+Copyright: 2023 MLabs ltd and cardano.nix contributors 
+License: Apache-2.0
+Files: *.lock *.nix *.md
```
Comment (aciceri) [inline .reuse/dep5:8]: If I understood it correctly from [here](https://reuse.software/spec/) or you put a comment at the beginning of files or you use this file to specify multiple sources at the same time.
Personally I prefer the latter way.

I can't understand what it your reasoning behind doing a mix of both. For instance why the you manually added the string to the `yaml` file?

Also, does it make sense licensing lockfiles or `.gitignore`s?

**Review by aciceri — CHANGES_REQUESTED** [review]
Could you please rename commits following conventional commits as usual? (I also always forget and then have to rename, I'm starting to doubt how useful they really are in our case)

File: .reuse/dep5 (hunk 1 above)
Comment (avnik) [reply .reuse/dep5]: done

File: .reuse/dep5 (hunk 2 above)
Comment (avnik) [reply .reuse/dep5]: I moved all to dep5, so all lint check passed, also add reuse tool to devshell

**Review by aciceri — APPROVED** [review]

**Pull request merged**